src/SamplingSchemes.jl

```diff
@@ -0,0 +1,314 @@
+        iter = liveset.walkers[1].iter
+    else
+        @warn "Failed to accept MC move"
+        emax = missing
+        ns_params.fail_count += 1
+    end
+    adjust_step_size(ns_params, rate)
+    return iter, emax, liveset, ns_params
+end
+
+"""
+    nested_sampling_step!(liveset::AtomWalkers, ns_params::NestedSamplingParameters, mc_routine::MCDemonWalk)
+
+Perform a single step of the nested sampling algorithm using the Monte Carlo demon walk routine.
+
+# Arguments
+- `liveset::AtomWalkers`: The set of atom walkers representing the current state of the system.
+- `ns_params::NestedSamplingParameters`: The parameters for the nested sampling algorithm.
+- `mc_routine::MCDemonWalk`: The parameters for the Monte Carlo demon walk.
+
+# Returns
+- `iter`: The iteration number after the step.
+- `emax`: The maximum energy recorded during the step.
+- `liveset`: The updated set of atom walkers after the step.
+- `ns_params`: The updated nested sampling parameters after the step.
+"""
+function nested_sampling_step!(liveset::AtomWalkers, ns_params::NestedSamplingParameters, mc_routine::MCDemonWalk)
+    sort_by_energy!(liveset)
+    ats = liveset.walkers
+    lj = liveset.lj_potential
+    iter::Union{Missing,Int} = missing
+    emax::Union{Missing,typeof(0.0u"eV")} = liveset.walkers[1].energy
+    pick_a_random_walker = rand(ats[2:end])
+    to_walk = deepcopy(pick_a_random_walker)
+    accept, rate, at, _, _ = MC_nve_walk!(ns_params.mc_steps, to_walk, lj, ns_params.step_size;
+                                        e_demon_tolerance=mc_routine.e_demon_tolerance,
+                                        demon_energy_threshold=mc_routine.demon_energy_threshold,
+                                        demon_gain_threshold=mc_routine.demon_gain_threshold,
+                                        max_add_steps=mc_routine.max_add_steps)
+    @info "acceptance rate: $rate, emax: $emax, is_accepted: $accept"
+    if accept
+        push!(ats, at)
+        popfirst!(ats)
+        update_iter!(liveset)
+        ns_params.fail_count = 0
+        iter = liveset.walkers[1].iter
+    else
+        @warn "Failed to accept MC move"
+        emax = missing
+        ns_params.fail_count += 1
+    end
+    adjust_step_size(ns_params, rate)
+    return iter, emax, liveset, ns_params
+end
+
+"""
+    adjust_step_size(ns_params::NestedSamplingParameters, rate::Float64)
+
+Adjusts the step size of the nested sampling algorithm based on the acceptance rate. 
+    If the acceptance rate is greater than 0.75, the step size is increased by 1%. 
+    If the acceptance rate is less than 0.25, the step size is decreased by 1%.
+
+# Arguments
+- `ns_params::NestedSamplingParameters`: The parameters of the nested sampling algorithm.
+- `rate::Float64`: The acceptance rate of the algorithm.
+
+# Returns
+- `ns_params::NestedSamplingParameters`: The updated parameters with adjusted step size.
+"""
+function adjust_step_size(ns_params::NestedSamplingParameters, rate::Float64)
```

Comment:
It doesn't seem like the user can select the range of acceptable rates.

Reply:
Good call! There should be an option.